code/modules/assembly/infrared.dm

```diff
@@ -246,7 +246,9 @@
 	hit()
 
 /obj/effect/beam/i_beam/Crossed(atom/movable/AM as mob|obj)
-	if(istype(AM, /obj/effect/beam) || !AM.density)
+	if(!isobj(AM) && !isliving(AM))
+		return
+	if(istype(AM, /obj/effect/beam))

```

Comment:
That would be more *realistic,* but it's probably not worth the mess of trying to differentiate them for the rare occasion it would matter.